### PR TITLE
types: fix compile warning about any when set strict:true in tsconfig.json

### PR DESCRIPTION
**Pull request by waitingsong**

### CODE REVIEW

File: ionic-angular/official/super/src/models/item.ts [hunk 1]
```diff
@@ -12,8 +12,13 @@ export class Item {
   constructor(fields: any) {
     // Quick and dirty extend/assign fields to this model
     for (const f in fields) {
+      // @ts-ignore
```
Comment (dwieeb) [inline ionic-angular/official/super/src/models/item.ts]: Why is this needed? The `Item` type should merge class and interface and allow an index signature.

Comment (waitingsong) [reply ionic-angular/official/super/src/models/item.ts]: It's unnecessary. At first i want to suppress the signature warning by this (but only for typescrpt v2.6+).

Comment (dwieeb) [reply ionic-angular/official/super/src/models/item.ts]: Also, isn't `@ts-ignore` only for `.js` files?

Comment (waitingsong) [reply ionic-angular/official/super/src/models/item.ts]: no. it is for ts  only typescript 2.6+ ,  er.. ionic lock ts version at 2.4.2 now.
1. only use @ts-ignore without definition of interface Item under typescrpt 2.6+
2. declare interface Item without @ts-ignore usually (under typescript 2.4.2)
